ios/Classes/DescopePlugin.swift

```diff
@@ -77,45 +80,71 @@ public class DescopePlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
     }
     
     // Internal
-    
+
+    @MainActor
     private func startFlow(_ urlString: String) {
-        Task { @MainActor in
-            guard var url = URL(string: urlString) else { return }
-            let session = ASWebAuthenticationSession(url: url, callbackURLScheme: redirectScheme) { callbackURL, error in
-                if let error {
-                    switch error {
-                    case ASWebAuthenticationSessionError.canceledLogin:
-                        self.eventSink?("canceled")
-                        return
-                    case ASWebAuthenticationSessionError.presentationContextInvalid, ASWebAuthenticationSessionError.presentationContextNotProvided:
-                        // not handled for now
-                        fallthrough
-                    default:
-                        self.eventSink?("")
-                        return
-                    }
+        startFlow(urlString, attempts: 5)
+    }
+
+    @MainActor
+    private func startFlow(_ urlString: String, attempts: Int) {
+        if attempts > 0, defaultContextProvider.findKeyWindow() == nil {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) { [self] in
+                startFlow(urlString, attempts: attempts - 1)
+            }
+            return
+        }
+        
+        guard let url = URL(string: urlString) else { return }
+        let session = ASWebAuthenticationSession(url: url, callbackURLScheme: redirectScheme) { [self] callbackURL, error in
+            if let error {
+                switch error {
+                case ASWebAuthenticationSessionError.canceledLogin:
+                    respond(with: "canceled")
+                    return
+                case ASWebAuthenticationSessionError.presentationContextInvalid,
+                    ASWebAuthenticationSessionError.presentationContextNotProvided:
+                    // not handled for now
+                    fallthrough
+                default:
+                    respond(with: "")
+                    return
                 }
-                self.eventSink?(callbackURL?.absoluteString ?? "")
             }
-            session.prefersEphemeralWebBrowserSession = true
-            session.presentationContextProvider = defaultContextProvider
-            session.start()
+            respond(with: callbackURL?.absoluteString ?? "")
         }
+        session.prefersEphemeralWebBrowserSession = true
+        session.presentationContextProvider = defaultContextProvider
+        sessions += [session]
+        session.start()
+    }
+    
+    @MainActor
+    private func respond(with response: String) {
+        eventSink?(response)
+        for session in sessions {
+            session.cancel()
```

Comment:
It's probably fine, but we're also cancelling the session that succeeded here

Reply:
True, but testing shows it's has no effect once the callback gets called